### PR TITLE
Add Kafka initial documetations

### PR DESCRIPTION
**Pull request by raihankhan**
- Add docs for kafka provisioning
- Add docs for kafka crd, version crd
- Add docs for features and demo

Signed-off-by: raihankhan <raihan@appscode.com>

### CODE REVIEW

File: docs/guides/kafka/clustering/combined-cluster/index.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+kafka-multinode   kubedb.com/v1alpha2   3.3.2     Provisioning   18s
+kafka-multinode   kubedb.com/v1alpha2   3.3.2     Provisioning   2m6s
+kafka-multinode   kubedb.com/v1alpha2   3.3.2     Provisioning   2m8s
+kafka-multinode   kubedb.com/v1alpha2   3.3.2     Ready          2m14s
+```
+
+Hence, the cluster is ready to use.
+Let's check the k8s resources created by the operator on the deployment of Kafka CRO:
+
+```bash
```
Comment (AbdullahAlShaad) [inline docs/guides/kafka/clustering/combined-cluster/index.md:153]: $ kubectl get all,secret,pvc -n demo -l 'app.kubernetes.io/instance=kafka-multinode'

File: docs/guides/kafka/clustering/topology-cluster/index.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+> Note: YAML files used in this tutorial are stored in [here](https://github.com/kubedb/docs/tree/{{< param "info.version" >}}/docs/examples/kafka/clustering) in GitHub repository [kubedb/docs](https://github.com/kubedb/docs).
+
+## Create Topology Kafka Cluster
+
+Here, we are going to create a TLS secured Kafka topology cluster in Kraft mode.
+
+### Create Issuer/ ClusterIssuer
+
+At first, make sure you have cert-manager installed on your k8s for enabling TLS. KubeDB operator uses cert manager to inject certificates into kubernetes secret & uses them for secure `SASL` encrypted communication among kafka brokers and controllers. We are going to create an example `Issuer` that will be used throughout the duration of this tutorial to enable SSL/TLS in Kafka. Alternatively, you can follow this [cert-manager tutorial](https://cert-manager.io/docs/configuration/ca/) to create your own `Issuer`.
```
Comment (AbdullahAlShaad) [inline docs/guides/kafka/clustering/topology-cluster/index.md:44]: secure SSL encrypted communication

File: docs/guides/kafka/concepts/kafka.md [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+  name: kf-auth
+  namespace: demo
+type: Opaque
+```
+
+Secrets provided by users are not managed by KubeDB, and therefore, won't be modified or garbage collected by the KubeDB operator (version 0.13.0 and higher).
+
+### spec.topology
+
+`spec.shardTopology` represents the topology configuration for Kafka cluster in KRaft mode.
```
Comment (AbdullahAlShaad) [inline docs/guides/kafka/concepts/kafka.md]: spec.topology

File: docs/guides/kafka/concepts/kafka.md [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+
+#### spec.podTemplate.spec.args
+
+`spec.podTemplate.spec.args` is an optional field. This can be used to provide additional arguments to database installation. To learn about available args of `mongod`, visit [here](https://docs.mongodb.com/manual/reference/program/mongod/).
+
+#### spec.podTemplate.spec.env
+
+`spec.podTemplate.spec.env` is an optional field that specifies the environment variables to pass to the MongoDB docker image. To know about supported environment variables, please visit [here](https://hub.docker.com/r/_/mongo/).
+
+Note that, KubeDB does not allow `MONGO_INITDB_ROOT_USERNAME` and `MONGO_INITDB_ROOT_PASSWORD` environment variables to set in `spec.podTemplate.spec.env`. If you want to use custom superuser and password, please use `spec.authSecret` instead described earlier.
```
Comment (AbdullahAlShaad) [inline docs/guides/kafka/concepts/kafka.md]: Remove MongoDB part

**Review by AbdullahAlShaad — CHANGES_REQUESTED** [review]

File: docs/guides/kafka/clustering/topology-cluster/index.md (hunk 2 above)
Comment (raihankhan) [reply docs/guides/kafka/clustering/topology-cluster/index.md:44]: It's actually `SASL` a mechanism Kafka uses for enabling TLS

**Review by AbdullahAlShaad — DISMISSED** [review]
LGTM!

**Comment by raihankhan** [comment]
> LGTM!

.

**Comment by raihankhan** [comment]
> LGTM!

Thanks. Once the UI is ready we will be able to view the demo preview.

**Pull request merged**